### PR TITLE
Add block0 golden tests from jormungandr-lib

### PR DESCRIPTION
**Pull request by Anviking**
# Issue Number

Related to #631, but not necessarily included there

# Overview

- [x] I have generated arbitrary genesis-configurations with jormungandr, and written the corresponding genesis-block-binaries to disk as goldens for the binary decoders.
- [x] I have added the `jormungandr`-diff with small instructions on how to run.

# Comments

- This gives us better coverage of tests for the binary format. Exactly to what extend I am not sure, but I had a look at the printed rust-objects corresponding to the config.
- Also, notably, the genesis-blocks have no transaction inputs.
- This fails without the #631 fix, and succeeds with it.

### CODE REVIEW

File: lib/jormungandr/test/unit/Cardano/Wallet/Jormungandr/BinarySpec.hs [hunk 1]
```diff
@@ -240,6 +247,17 @@ spec = do
             res <- try' (runGet getAddress (BL.fromStrict $ unsafeFromHex hex))
             res `shouldSatisfy` isRight
 
+        describe "golden block0s from jcli" $ do
```
Comment (Anviking) [inline lib/jormungandr/test/unit/Cardano/Wallet/Jormungandr/BinarySpec.hs]: Ok, `jcli` is probably not the best term here. We are using an instance in `jormungandr-lib`, so "jormungandr-lib" would probably be more appropriate.

File: lib/jormungandr/test/unit/Cardano/Wallet/Jormungandr/BinarySpec.hs [hunk 2]
```diff
@@ -240,6 +247,17 @@ spec = do
...
+            let dir = "test/data/block0s"
```
Comment (Anviking) [inline lib/jormungandr/test/unit/Cardano/Wallet/Jormungandr/BinarySpec.hs:251]: I was lazy. To be consistent this should probably be `test/data/Cardano/Wallet/Jormungandr/BinarySpec/block0s`.

The term "block0" vs "genesis-block" could also be debated.

File: lib/jormungandr/test/unit/Cardano/Wallet/Jormungandr/BinarySpec.hs (hunk 1 above)
Comment (paweljakubas) [reply lib/jormungandr/test/unit/Cardano/Wallet/Jormungandr/BinarySpec.hs]: you produced them using the rust test in `jormungandr-lib` as specified in  `lib/jormungandr/test/data/block0s/jormungandr.patch ` above. So I believe the description should go :
`golden block0s generated from jormungandr-lib`

File: lib/jormungandr/test/data/block0s/jormungandr.patch [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+--- a/jormungandr-lib/src/interfaces/block0_configuration/mod.rs
++++ b/jormungandr-lib/src/interfaces/block0_configuration/mod.rs
+@@ -132,6 +132,33 @@ mod test {
+         }
+     }
+ 
++    #[test]
++    // Can be run with:
++    // $ cd jormungandr-lib
++    // $ cargo test write_arbitrary_block0s_to_disk
```
Comment (paweljakubas) [inline lib/jormungandr/test/data/block0s/jormungandr.patch:12]: :ok_hand:

File: lib/jormungandr/cardano-wallet-jormungandr.cabal [hunk 4]
```diff
@@ -89,6 +89,7 @@ test-suite unit
     , cardano-crypto
     , cardano-wallet-jormungandr
     , containers
+    , directory
```
Comment (paweljakubas) [inline lib/jormungandr/cardano-wallet-jormungandr.cabal:92]: we will need corresponding `nix` entry for this in `https://github.com/input-output-hk/cardano-wallet/blob/master/nix/.stack.nix/cardano-wallet-jormungandr.nix`

**Review by paweljakubas — APPROVED** [review]
LGTM! modulo nix and maybe refining test description

File: lib/jormungandr/test/unit/Cardano/Wallet/Jormungandr/BinarySpec.hs (hunk 1 above)
Comment (paweljakubas) [reply lib/jormungandr/test/unit/Cardano/Wallet/Jormungandr/BinarySpec.hs]: changed in https://github.com/input-output-hk/cardano-wallet/pull/633/commits/40766eeeef457a9fa4c9f0a641df6b9f02a9219e

File: lib/jormungandr/cardano-wallet-jormungandr.cabal (hunk 4 above)
Comment (paweljakubas) [reply lib/jormungandr/cardano-wallet-jormungandr.cabal:92]: adopted in https://github.com/input-output-hk/cardano-wallet/pull/633/commits/40766eeeef457a9fa4c9f0a641df6b9f02a9219e

**Pull request merged**